pydantic/_internal/_decorators.py

```diff
@@ -513,6 +513,14 @@ def build(model_dc: type[Any]) -> DecoratorInfos:  # noqa: C901 (ignore complexi
             model_dc.__pydantic_decorators__ = res
             for name, value in to_replace:
                 setattr(model_dc, name, value)
+        if hasattr(model_dc, '__pydantic_decorators__'):
+            is_overridden, duplicated_methods = is_method_overridden(model_dc)
+            if is_overridden:
+                raise PydanticUserError(
+                    f'this method {duplicated_methods} is overriding existing method',
```

Comment:
Let's phrase this `Validator method {duplicated_methods} overrides existing validator method, this is not allowed`.
And if there are multiple duplicates, let's have a separate message listing them all.